### ISSUE TITLE
clippy::needless_borrows_for_generic_args fails to notice usage in closure

### ISSUE THREAD

**Issue by SUPERCILEX** [rust-lang/rust-clippy#13095]
### Summary

```rust
                    create_dir_all(&file)
                        .attach_printable_lazy(|| format!("Failed to create directory {file:?}"))?;
```

An attempt is make to replace `&file` with `file` which doesn't work since `file` is later moved into the closure.

### Lint Name

needless_borrows_for_generic_args

### Reproducer

Full project here: https://github.com/SUPERCILEX/ftzz/blob/a8a6b0d5151a13df44ac9c8ab64b6439b7d4b5a3/src/core/files.rs#L95

### Version

```text
rustc 1.81.0-nightly (fcaa6fdfb 2024-07-13)
binary: rustc
commit-hash: fcaa6fdfbee1316184e7ad98c53241d52cd30a5f
commit-date: 2024-07-13
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```

**Comment by SUPERCILEX** [comment]
Actually it turns out that you just need to use `&*` instead of `&`.
